config/routes.rb

```diff
@@ -154,6 +118,40 @@
     resources :sites
   end
 
+  get '/attachment/:id/download(/:name)' => 'attachments#download',
+      as: :download_attachment
+  get '/attachment/:id/show' => 'attachments#show',
+      as: :show_attachment
+
+  # Picture urls
+  get "/pictures/:id/show(/:size)(/:crop)(/:crop_from/:crop_size)(/:quality)/:name.:format" => 'pictures#show',
+      as: :show_picture
+  get '/pictures/:id/zoom/:name.:format' => 'pictures#zoom',
+      as: :zoom_picture
+  get "/pictures/:id/thumbnails(/:size)(/:crop)(/:crop_from/:crop_size)/:name.:format" => 'pictures#thumbnail',
+      as: :thumbnail, :defaults => {:format => 'png', :name => "thumbnail"}
+
+  resources :messages, :only => [:index, :new, :create]
+  resources :elements, :only => :show
+  resources :contents, :only => :show
```

Comment:
Style/HashSyntax: Use the new Ruby 1.9 hash syntax. (https://github.com/bbatsov/ruby-style-guide#hash-literals)